Mage.Sets/src/mage/cards/m/MyojinOfCrypticDreams.java

```diff
@@ -0,0 +1,78 @@
+package mage.cards.m;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.EntersBattlefieldAbility;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.condition.common.CastFromHandSourcePermanentCondition;
+import mage.abilities.costs.common.RemoveCountersSourceCost;
+import mage.abilities.decorator.ConditionalOneShotEffect;
+import mage.abilities.effects.common.CopyTargetSpellEffect;
+import mage.abilities.effects.common.counter.AddCountersSourceEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.constants.TargetController;
+import mage.counters.CounterType;
+import mage.filter.FilterSpell;
+import mage.filter.predicate.Predicate;
+import mage.game.Game;
+import mage.game.stack.StackObject;
+import mage.target.TargetSpell;
+
+import java.util.UUID;
+
+/**
+ * @author Alex-Vasile
+ */
+public class MyojinOfCrypticDreams extends CardImpl {
+
+    private static final FilterSpell permanentSpellFilter = new FilterSpell("permanent spell you control");
+    static {
+        permanentSpellFilter.add(TargetController.YOU.getControllerPredicate());
+        permanentSpellFilter.add(MyojinOfCrypticDreamsPredicate.instance);
+    }
+
+    public MyojinOfCrypticDreams(UUID ownderId, CardSetInfo setInfo) {
+        super(ownderId, setInfo, new CardType[]{CardType.CREATURE}, "{5}{U}{U}{U}");
+
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.SPIRIT);
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(3);
+
+        // Myojin of Cryptic Dreams enters the battlefield with an indestructible counter on it if you cast it from your hand.
+        this.addAbility(new EntersBattlefieldAbility(new ConditionalOneShotEffect(
+                new AddCountersSourceEffect(CounterType.INDESTRUCTIBLE.createInstance()),
```

Comment:
Neither this nor the other Myojin from NEO seem to have an indicator for the number of indestructible counters on it.

Reply:
I don't understand, what kind of indicator do you mean?

Reply:
Is there not supposed to be an indicator on the card showing that it has an indestructible counter on it?

Reply:
oh that might be another graphical bug, I'm not sure how the client handles ability counters